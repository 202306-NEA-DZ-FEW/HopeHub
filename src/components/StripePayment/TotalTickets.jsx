import React, { useEffect, useState } from "react";
import { collection, query, where, getDocs } from "firebase/firestore";
import { db } from "@/util/firebase";

const TotalTickets = ({ user }) => {
    const [totalTicketsFromPayments, setTotalTicketsFromPayments] = useState(0);

    useEffect(() => {
        const fetchTotalTicketsFromPayments = async () => {
            try {
                if (user) {
                    const paymentsCollectionRef = collection(
                        db,
                        "customers",
                        user.uid,
                        "payments"
                    );

                    const paymentsQuerySnapshot = await getDocs(
                        query(
                            paymentsCollectionRef,
                            where("status", "==", "succeeded")
                        )
                    );

                    let totalTickets = 0;

                    for (const paymentDoc of paymentsQuerySnapshot.docs) {
                        const paymentData = paymentDoc.data();
                        const itemsArray = paymentData.items || [];

                        itemsArray.forEach((item) => {
                            const description = item.description || "";
                            const ticketsMatch =
                                description.match(/(\d+) Tickets/);

                            if (ticketsMatch) {
                                const ticketsCount = parseInt(
                                    ticketsMatch[1],
                                    10
                                );
                                totalTickets += ticketsCount;
                            }
                        });
                    }

                    setTotalTicketsFromPayments(totalTickets);
                }
            } catch (error) {
                console.error(error);
            }
        };

        fetchTotalTicketsFromPayments();
    }, [user]);

    return <div>Total Tickets from Payments: {totalTicketsFromPayments}</div>;
};

export default TotalTickets;
