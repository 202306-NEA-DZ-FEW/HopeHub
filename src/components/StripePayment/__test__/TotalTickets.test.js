import renderer from "react-test-renderer";

import TotalTickets from "../TotalTickets";

it("renders correctly", () => {
    const tree = renderer.create(<TotalTickets />).toJSON();
    expect(tree).toMatchSnapshot();
});
