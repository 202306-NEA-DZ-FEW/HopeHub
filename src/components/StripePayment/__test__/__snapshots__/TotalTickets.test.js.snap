// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="font-poppins text-base mx-6 md:text-xl md:mx-9 md:pb-9 pb-6 text-NeutralBlack dark:text-NeutralWhite "
>
  Total Tickets: 
  0
</div>
`;
